### DISCUSSION TITLE
How to console the row and column in time when cursor locate in the editor? Is there an official explanation?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [microsoft/monaco-editor#4621] [Q&A] [answered]
I have a monaco-editor demo as below: 

https://github.com/liaodalin19903/Test-demos/tree/use-001-howtogetcurrentcursor/react-test-demos/react-demo-01-monaco-editor  

1、The function I want to implement is to click the mouse in the editor, and then print the row and column in time(such as in the snapshot:  print out: `9, 6` ). How to do that? Is there an official explanation?

![image](https://github.com/user-attachments/assets/3dcf6a8f-cc63-4b4c-8712-fa083c203046)

2、
Is it possible to get inside a method/property, class, or expression that the mouse cursor is in?  
such as there can get  the expression `add(1, 11)`

**Comment by rcjsuen** [comment]
> The function I want to implement is to click the mouse in the editor, and then console the row and column in time.

What do you mean by "console the row and column in time"? 🤔

**Comment by liaodalin19903** [comment]
i mean print the line number and the index number at the line.



---- Replied Message ----
| From | Remy ***@***.***> |
| Date | 07/30/2024 02:30 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [microsoft/monaco-editor] How to console the row and column in time when cursor locate in the editor? Is there an official explanation? (Discussion #4621) |

The function I want to implement is to click the mouse in the editor, and then console the row and column in time.

What do you mean by "console the row and column in time"? 🤔

**Comment by rcjsuen** [comment, marked as answer]
@liaodalin19903 Perhaps [this snippet of code](https://microsoft.github.io/monaco-editor/playground.html?source=v0.50.0#XQAAAAKuAQAAAAAAAABBqQkHQ5NjdPr5qlU4WDmTRXNuTfnoBAhm7hasbKCuA-Xn2CWEcQNiZX9dRgB3KMlME_xfJcBLZBOS8iExS4SGu7omJ7oYdTK94nbK9eCs3q-E7jR31t4yrT4cczCSXkHYQPL7BEtBfNslHRqg3g9yzUlMj-BnpKmiyWwKTwmh940cfm_lgHN0yhG1HTlfahVfPNEnTarpveeGCcwdcosvJrJqe-jIrzGVxPmSw1M0cmafWEY1XmPxeNcdiqH7dTZe8vEk6CfKFjtNBX5G9Bm_Bcph6qVUgyQ2Z3TAfSaV-QZKcrv0wLsBxyMuoET0ruYu5MHrvuSPTaIVWxT0IiqurZAmKKWn202nUK8uxtw_8FnvPENa8W5ILvsOX-fOm4zu7E-X5Famj6u8QDhgzj2j_Ef-S_Vt) may be of help to you then.

```JavaScript
const value = `function hello() {
	alert('Hello world!');
}`;

const editor = monaco.editor.create(
	document.getElementById("container"), {
	value,
	language: "javascript",
	automaticLayout: true,
});

editor.onDidChangeCursorPosition((e) => {
	document.getElementById("console").textContent = JSON.stringify(e.position);
});
```
```HTML
<div id="container" style="height: 100%">
    <div id="console">Console</div>
</div>
```
